### ISSUE TITLE
[bug] productName is now mandatory

### ISSUE THREAD

**Issue by mihai-dinculescu** [tauri-apps/tauri#5601]
### Describe the bug

The documentation says that `productName` is supposed to be optional (`string?`), but the new regex introduced in https://www.github.com/tauri-apps/tauri/commit/b9316a64eaa9348c79efafb8b94960d9b4d5b27a makes it mandatory.

### Reproduction

1. Define a tauri.conf.json without `package` or without `package.productName`.
2. `cargo tauri build --verbose`

### Expected behavior

Build succeeds

### Platform and versions

```bash
Environment
  › OS: Windows 10.0.20348 X64
  › Webview2: 107.0.1418.35
  › MSVC: 
      - Visual Studio Enterprise 2022
  › Node.js: 16.18.0
  › npm: 8.19.2
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.65.0
  › cargo: 1.65.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.1.1
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.1,
  › tauri-build [RUST]: 1.1.1,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.1,

App
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Malformed { path: "definitions/PackageConfig/properties/productName", detail: "The value of pattern MUST be a valid RegExp, but Syntax(\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\nregex parse error:\n    ^[^/\\:*?\"<>|]+$\n        ^^\nerror: unrecognized escape sequence\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n)" }', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\tauri-cli-1.1.1\src\helpers\config.rs:139:58
```

**Comment by FabianLars** [comment]
Hmm, it still works fine for me without the `package` object. Do you see any error messages?

**Comment by mihai-dinculescu** [comment]
I'm sorry, I've added the wrong `tauri info` initially. Here's the correct one. As you can notice, it's on 1.1.1 at the time of the failure.

```
Environment
  › OS: Windows 10.0.20348 X64
  › Webview2: 107.0.1418.35
  › MSVC: 
      - Visual Studio Enterprise 2022
  › Node.js: 16.18.0
  › npm: 8.19.2
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.1
  › rustc: 1.65.0
  › cargo: 1.65.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.1.1
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.1,
  › tauri-build [RUST]: 1.1.1,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.1,

App
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Malformed { path: "definitions/PackageConfig/properties/productName", detail: "The value of pattern MUST be a valid RegExp, but Syntax(\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\nregex parse error:\n    ^[^/\\:*?\"<>|]+$\n        ^^\nerror: unrecognized escape sequence\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n)" }', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\tauri-cli-1.1.1\src\helpers\config.rs:139:58
```

I've got this in the CI in a public repo https://github.com/mihai-dinculescu/msfs-2020-gps-link/actions/runs/3432127585/jobs/5721057788.

After updating to 1.2.0 across the board, the error went away. Here's the build: https://github.com/mihai-dinculescu/msfs-2020-gps-link/actions/runs/3432146924/jobs/5721101500.

The interesting part is that two days ago, the same code and config passed on the very same 1.1.1 version. Unfortunately, I didn't have `tauri info` in the CI then, but here's the successful build anyway: https://github.com/mihai-dinculescu/msfs-2020-gps-link/actions/runs/3413817752/jobs/5680981498.

**Comment by parker-codes** [comment]
I just ran into the same issue! After following the stacktrace's clues across the tauri-cli code without success, I discussed the issue with Lucas. Here's what he found: 
> the cli has a tauri dependency (tauri-utils) that breaks if you update it without updating everything else
> 
> cli@1.1.1 used tauri-utils@1.1, but it broke when we released tauri-utils@1.2

You can lock `tauri-cli`'s dependencies so that the versions align properly. For example, in my Dockerfile I was running `RUN cargo install tauri-cli --locked --version 1.1.1 --force`. I added `--locked` so that it becomes `RUN cargo install tauri-cli --locked --version 1.1.1 --force`.

**Comment by lucasfernog** [comment]
I'll pin tauri-utils for each minor version but it needs https://github.com/jbolda/covector/pull/252
